go/src/koding/klient/app/defertime.go

```diff
@@ -0,0 +1,82 @@
+package app
+
+import "time"
+
+// DeferTime allows to invoke registered function after provided duration. It
+// can cancel deferred function call even if internal timer was already started.
+type DeferTime struct {
```

Comment:
Yes, you are right, I'm leaking 1 go-routine which is meant to live during the entire klient's life-cycle. I wanted to add `Stop()` method but as we leak other goroutines that also have to be open during klient life, i decided to not complicate this struct. I see no point in it.

@cihangir I can inline this logic in klient, but I won't be able to test it easily. If you let me to leave it, I will add `Exit()` method which will stop *leaked* go-routine.